src/System.Collections/src/System/Collections/Generic/Queue.cs

```diff
@@ -181,17 +181,36 @@ void ICollection.CopyTo(Array array, int index)
         }
 
         // Adds item to the tail of the queue.
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public void Enqueue(T item)
         {
-            if (_size == _array.Length)
+            int size = _size;
+            T[] array = _array;
+
+            if ((uint)size < (uint)array.Length)
             {
-                int newcapacity = (int)((long)_array.Length * (long)GrowFactor / 100);
-                if (newcapacity < _array.Length + MinimumGrow)
-                {
-                    newcapacity = _array.Length + MinimumGrow;
-                }
-                SetCapacity(newcapacity);
+                _array[_tail] = item;
```

Comment:
```suggestion
                array[_tail] = item;
```
Use the local copy -> saves one `mov      rcx, rdx` (although the cpu may execute this at zero latency, it has the cost to fetch and decode).

Reply:
Although the whole could be written as:
```c#
public void Enqueue(T item)
{
    int tail = _tail;
    int size = _size;
    T[] array = _array;

    if ((uint)size < (uint)array.Length && (uint)tail < (uint)array.Length)
    {
        array[tail] = item;
        MoveNext(ref tail, array);
        _tail = tail;
        _size = size + 1;
        _version++;
    }
    else
    {
        EnqueueWithResize(item);
    }
}

// ...

private void MoveNext(ref int index) => MoveNext(ref index, _array);

// Increments the index wrapping it if necessary.
private void MoveNext(ref int index, T[] array)
{
    // It is tempting to use the remainder operator here but it is actually much slower
    // than a simple comparison and a rarely taken branch.
    // JIT produces better code than with ternary operator ?:
    int tmp = index + 1;
    if (tmp == array.Length)
    {
        tmp = 0;
    }
    index = tmp;
}
```
So in `MoveNext` values that are in registers can be reused, so eliminating some memory accesses.

Reply:
Is `int MoveNext (int index)` faster than `void MoveNext(ref int index)`? I have seen this in other contexts though msybe for reference types.

Reply:
Here it is equal fast, as it gets inlined.
When not inlined, `int MoveNext(int)` may be faster, as args as well ret are in registers, as opposed to stack in the `ref arg`-case.

Reply:
I've made these updates